File: src/main/java/no/nav/pensjon/selvbetjeningopptjening/opptjening/mapping/AfpHistorikkMapper.java
package no.nav.pensjon.selvbetjeningopptjening.opptjening.mapping;

import no.nav.pensjon.selvbetjeningopptjening.model.AfpHistorikkDto;
import no.nav.pensjon.selvbetjeningopptjening.opptjening.AfpHistorikk;

public class AfpHistorikkMapper {

    public static AfpHistorikk fromDto(AfpHistorikkDto dto) {
        return dto == null ? null
                :
                new AfpHistorikk(
                        dto.getVirkFom(),
                        dto.getVirkTom());
    }
}

File: src/main/java/no/nav/pensjon/selvbetjeningopptjening/opptjening/mapping/AfpHistorikkMapper.kt
package no.nav.pensjon.selvbetjeningopptjening.opptjening.mapping

import no.nav.pensjon.selvbetjeningopptjening.model.AfpHistorikkDto
import no.nav.pensjon.selvbetjeningopptjening.opptjening.AfpHistorikk

object AfpHistorikkMapper {

    fun fromDto(dto: AfpHistorikkDto?): AfpHistorikk? =
        dto?.let {
            AfpHistorikk(
                virkningFomDate = it.virkFom,
                virkningTomDate = it.virkTom
            )
        }
}